pkg/controller/baremetalhost/baremetalhost_controller.go

```diff
@@ -43,11 +45,22 @@ const (
 
 var runInTestMode bool
 var runInDemoMode bool
+var maxConcurrentReconciles int = 1
 
 func init() {
 	flag.BoolVar(&runInTestMode, "test-mode", false, "disable ironic communication")
 	flag.BoolVar(&runInDemoMode, "demo-mode", false,
 		"use the demo provisioner to set host states")
+
+	var err error
```

Comment:
You don't need this if you use `:=` on the line below

Reply:
Actually using := will locally scope the variables, including maxConcurrentReconciles. So when init() exits, the global maxConcurrentReconciles (with default 0) will take over. Hence the little work around.